src/tools.rs

```diff
@@ -185,3 +185,51 @@ pub fn parse_maps(pid: u32) -> Result<Vec<LinuxMapsEntry>, Error> {
 
     Ok(entries)
 }
+
+#[inline(always)]
+pub fn decode_uhalf(data: &[u8]) -> u16 {
+    (data[0] as u16) | ((data[1] as u16) << 8)
+}
+
+#[allow(dead_code)]
+#[inline(always)]
+pub fn decode_shalf(data: &[u8]) -> i16 {
+    let uh = decode_uhalf(data);
+    if uh >= 0x8000 {
+        ((uh as i32) - 0x10000) as i16
+    } else {
+        uh as i16
+    }
+}
+
+#[inline(always)]
+pub fn decode_uword(data: &[u8]) -> u32 {
+    (data[0] as u32) | ((data[1] as u32) << 8) | ((data[2] as u32) << 16) | ((data[3] as u32) << 24)
```

Comment:
Documentation what this code is trying to do would go a long way. Would https://doc.rust-lang.org/std/primitive.u32.html#method.from_be_bytes or similar help?